modules/zenith-public/lua/4-additive/npc/jeuno/[LQS]bagotrix.lua

```diff
@@ -0,0 +1,162 @@
+    {
+        ['Upper_Jeuno'] =
+        {
+            {
+                name    = bagotrix,
+                type    = xi.objType.NPC,
+                look    = 85,
+                pos     = { 0.4745, 1.8001, 75.9075, 9 },
+                -- REQUIRED: default event table needed for onTrigger to be registered
+                default =
+                {
+                    'Psssst! Uplander! Bagotrix have special deal for you!',
+                },
+            },
+        },
+    },
+    steps =
+    {
+        -- Step 0: Main interaction - check and offer Mog Satchel
+        {
+            [bagotrix] =
+            {
+                onTrigger = function(player, npc)
+                    -- Check if already purchased/unlocked
+                    if hasSatchelUnlocked(player) then
+                        player:printToPlayer('Bagotrix: Oho! Uplander already have big satchel!', xi.msg.channel.NS_SAY, npc:getPacketName())
+                        player:printToPlayer(' Bagotrix no have more to sell you!', xi.msg.channel.NS_SAY)
+                        player:printToPlayer(' You keep satchel full of treasures, yes?', xi.msg.channel.NS_SAY)
+                        return
+                    end
+
+                    -- Calculate price
+                    local price, totalRanks = calculateSatchelPrice(player)
+
+                    -- Show initial greeting
+                    player:printToPlayer('Bagotrix: Psssst! Uplander! You want special deal?', xi.msg.channel.NS_SAY, npc:getPacketName())
+
+                    player:timer(1500, function(playerArg)
+                        playerArg:printToPlayer(' Bagotrix have Mog Satchel unlock!', xi.msg.channel.NS_SAY)
+                        playerArg:printToPlayer(' Very rare! Very special!', xi.msg.channel.NS_SAY)
+                    end)
+
+                    player:timer(3000, function(playerArg)
+                        playerArg:printToPlayer(string.format(' For you, only %s gil!', price), xi.msg.channel.NS_SAY)
+                        playerArg:printToPlayer(string.format(' You have %d completed mission ranks!', totalRanks), xi.msg.channel.NS_SAY)
+                        playerArg:printToPlayer(' Each rank save you 37,000 gil! Good deal, yes?', xi.msg.channel.NS_SAY)
+                    end)
+
+                    -- Show purchase menu
+                    player:timer(5000, function(playerArg)
+                        local menuPrice = calculateSatchelPrice(playerArg)
+
+                        playerArg:customMenu({
+                            title   = string.format('Purchase Mog Satchel for %s gil?', menuPrice),
+                            options =
+                            {
+                                {
+                                    'No thanks',
+                                    function()
+                                        playerArg:printToPlayer('Bagotrix: Hmph! Uplander no want good deal?', xi.msg.channel.NS_SAY, npc:getPacketName())
```

Comment:
I believe this function is executed with `PChar` as the first parameter. Try it out please with this change anytime you use `customMenu`

In general it's not good to use inherited lua variables that are pointers to CPP objects. If you want to get the npc's name, use `local npcName = npc:getPacketName()` and use `npcName` inside the anonymous function passed in `options`

```suggestion
                                    function(playerMenuArg)
                                        playerMenuArg:printToPlayer('Bagotrix: Hmph! Uplander no want good deal?', xi.msg.channel.NS_SAY)
```

Reply:
done